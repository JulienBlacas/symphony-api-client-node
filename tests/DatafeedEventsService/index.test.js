jest.mock('../../lib/SymConfigLoader', () => ({
  SymConfig: { keyAuthHost: 'keyauth.example.com' },
}))
jest.mock('../../lib/SymBotAuth', () => ({
  sessionAuthToken: 'session-token',
  kmAuthToken: 'key-manager-token',
}))
const https = require('https')
const nock = require('nock')
const DatafeedEventsService = require('../../lib/DatafeedEventsService')

describe('DatafeedEventsService', () => {
  beforeAll(() => {
    jest.spyOn(https, 'request')
  })

  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('#readDatafeed', () => {
    it.skip('handles error', () => {
      const id = 'abc123'
      nock('https://keyauth.example.com')
        .get(`/agent/v4/datafeed/${id}/read`)
        .reply(200, '{"status":"success"}')
      nock('https://keyauth.example.com')
        .get(`/agent/v4/datafeed/${id}/read`)
        .reply(200, '{"status":"test"}')

      expect(DatafeedEventsService.readDatafeed(id)).resolves.toBeUndefined()

      expect(https.request).toHaveBeenCalledTimes(2)
      expect(https.request.mock.calls).toMatchSnapshot()
    })
  })
})
