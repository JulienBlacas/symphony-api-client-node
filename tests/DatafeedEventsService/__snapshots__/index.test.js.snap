// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatafeedEventsService #readDatafeed handles error 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "keyManagerToken": "key-manager-token",
        "sessionToken": "session-token",
      },
      "host": "localhost:443",
      "hostname": "localhost",
      "method": "GET",
      "path": "/agent/v4/datafeed/abc123/read",
      "port": 443,
      "proto": "https",
    },
    [Function],
  ],
]
`;
